### ISSUE TITLE
Worker permanently closed during PHP-PM bootstrap.

### ISSUE THREAD

**Issue by rhclayto** [bertrama/php-pm-drupal-seven#6]
Trying to start this up I get the following error:

```
/var/www/hosting/example/com/htdocs/backend
+---------------------+------------------------+
| bridge              | DrupalSevenKernel      |
| host                | 127.0.0.1              |
| port                | 8080                   |
| workers             | 8                      |
| app-env             | dev                    |
| debug               | 1                      |
| logging             | 1                      |
| static              | 1                      |
| bootstrap           | DrupalSeven            |
| max-requests        | 1000                   |
| concurrent-requests |                        |
| socket-path         | .ppm/run/              |
| cgi-path            | /usr/local/bin/php-cgi |
+---------------------+------------------------+
Starting PHP-PM with 8 workers, using StreamSelectLoop ...
Worker permanently closed during PHP-PM bootstrap. Not so cool. Not your fault, please create a ticket at github.com/php-pm/php-pm withthe output of `ppm start -vv`.
Worker permanently closed during PHP-PM bootstrap. Not so cool. Not your fault, please create a ticket at github.com/php-pm/php-pm withthe output of `ppm start -vv`.
Worker permanently closed during PHP-PM bootstrap. Not so cool. Not your fault, please create a ticket at github.com/php-pm/php-pm withthe output of `ppm start -vv`.
Worker permanently closed during PHP-PM bootstrap. Not so cool. Not your fault, please create a ticket at github.com/php-pm/php-pm withthe output of `ppm start -vv`.
Worker permanently closed during PHP-PM bootstrap. Not so cool. Not your fault, please create a ticket at github.com/php-pm/php-pm withthe output of `ppm start -vv`.
Worker permanently closed during PHP-PM bootstrap. Not so cool. Not your fault, please create a ticket at github.com/php-pm/php-pm withthe output of `ppm start -vv`.
Worker permanently closed during PHP-PM bootstrap. Not so cool. Not your fault, please create a ticket at github.com/php-pm/php-pm withthe output of `ppm start -vv`.
Worker permanently closed during PHP-PM bootstrap. Not so cool. Not your fault, please create a ticket at github.com/php-pm/php-pm withthe output of `ppm start -vv`.
```

Running it with -vv gives me:

```
Starting PHP-PM with 8 workers, using StreamSelectLoop ...
Start new worker #5501
Start new worker #5502
Start new worker #5503
Start new worker #5504
Start new worker #5505
Start new worker #5506
Start new worker #5507
Start new worker #5508
```
plus the original error messages.

I know this is highly experimental, but I read that you had got it to the point of being able to serve a page request. Do you have any ideas what could be going on?

Regards.

**Comment by bertrama** [comment]
I did get this to serve a page successfully.  Serving two different pages has had mixed results depending on the use of globals in the code and dynamic blocks, etc.  And I never got as far as getting authentication to work.  I'm happy to retry it with a stock Drupal install and see if I can reproduce the behavior you see.

**Comment by bertrama** [comment]
I found my last test scenario: a stock drupal 7.43 install.  Attached are my composer.json and composer.lock files.  I put the composer files into the drupal root, and have the dependencies installed there.

I also had to update my php.ini to enable the pcntl functions, but that may be system dependent.

Here's the line I used to start it:

```bash
vendor/bin/ppm \
  start \
  . \
  --bridge=DrupalSevenKernel \
  --bootstrap=DrupalSeven \
  -vvv
```

[composer.json.txt](https://github.com/bertrama/php-pm-drupal-seven/files/837166/composer.json.txt)
[composer.lock.txt](https://github.com/bertrama/php-pm-drupal-seven/files/837167/composer.lock.txt)

**Comment by rhclayto** [comment]
Hi! Thanks for responding & for looking into this.

I was having a lot of problems with the PHP-PM components, so I decided to simplify by jettisoning PHP-PM & fuse React & Drupal together directly, using your project's approach as a guide, with success.

I have a script that bootstraps Drupal; sets up superglobals, etc.; starts a React loop & HTTP server; accepts HTTP requests (GET/POST/PATCH/OPTIONS tested so far) & runs them through Drupal & returns the responses to the client; & tears down the Drupal state at the end of the request, so as to be ready for the next one.

I am using the RESTful module to serve a REST API using Drupal, & am not doing any traditional page generation & serving with this hybrid React/Drupal daemon; in addition, I'm not using Drupal or PHP sessions at all, since I'm using RESTful's token authentication sub-module. I also had to patch RESTful in a couple of places to get it all working, but while I am sure there are still bugs to be hunted down, I am effectively using this hybrid to serve all requests I have been serving out of the stock Drupal/RESTful API with great success. I am using supervisord to manage the daemon processes.

Thanks for your work on this project, which gave me the insight needed to get this going.

Cheers,
H.
